solver/src/encoding.rs

```diff
@@ -1,101 +1,61 @@
-use std::convert::TryInto;
-
-use model::order::{OrderCreation, OrderKind};
-use primitive_types::U256;
-
-pub const TRADE_STRIDE: usize = 206;
+use model::{
+    order::{OrderCreation, OrderKind},
+    SigningScheme,
+};
+use primitive_types::{H160, U256};
+
+pub type EncodedTrade = (
+    U256,     // sellTokenIndex
+    U256,     // buyTokenIndex
+    H160,     // receiver
+    U256,     // sellAmount
+    U256,     // buyAmount
+    u32,      // validTo
+    [u8; 32], // appData
+    U256,     // feeAmount
+    U256,     // flags
+    U256,     // executedAmount
+    Vec<u8>,  // signature
+);
 
 /// Creates the data which the smart contract's `decodeTrade` expects.
 pub fn encode_trade(
     order: &OrderCreation,
-    sell_token_index: u8,
-    buy_token_index: u8,
+    sell_token_index: usize,
+    buy_token_index: usize,
```

Comment:
What was the purpose for this change? Does the `usize`hold an integer representation of the token's address? How was this a `u8` before?

Reply:
You can see in the contract's settlement code that the index is a U256 now. We are never going to have that many trades so I am using usize.